### PR TITLE
Issue2826: fix osx build error

### PR DESCRIPTION
**Pull request by persiaAziz**

### CODE REVIEW

**Comment by persiaAziz** [comment]
New commit: this patch by Alan worked on @d2r's mac

**Comment by persiaAziz** [comment]
[approve ci autest]

**Comment by SolidWallOfCode** [comment]
This builds on the standard PR systems, @bryancall or @zwoop needs to verify it builds on OSX. As noted, we tested on @d2r's MBP and it built there but would like another check.

**Comment by gtenev** [comment]
Removed only `-no-undefined` as a work-around and built successfully but then wondered why did we add it in the first place?

File: lib/Makefile.am [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-SUBDIRS = ts records tsconfig bindings cppapi
+SUBDIRS = ts records . tsconfig bindings cppapi
```
Comment (bryancall) [inline lib/Makefile.am]: This is not needed.  I tested the build without having this in and it worked.  Approving for now to get the build fixed.

**Comment by zwoop** [comment]
I think that was added by @jpeach as an optimization, not sure how important it is / was.

File: lib/Makefile.am (hunk 1 above)
Comment (SolidWallOfCode) [reply lib/Makefile.am]: Yeah, wasn't sure about that. The point of it was to force building the luajit before things that depend on it.

Comment (SolidWallOfCode) [reply lib/Makefile.am]: I think this may be needed for `make check`.

Comment (d2r) [reply lib/Makefile.am]: From what I recall, we attempted to fix the link issue first by having the lua code build first, but that was not the problem. 

This change may be neither sufficient nor necessary to fix the build.

**Comment by jpeach** [comment]
`-no-undefined` ensures that all the symbols are bound in the DSO at link time. Now you have a DSO that has symbol dependencies on something else. So `libtsconfig.so` is no longer standalone and can only be loaded by something that exports Lua symbols. Not sure whether that matters here, but FWIW.

**Pull request merged**